lib/kubernetes-deploy/kubernetes_resource/service.rb

```diff
@@ -16,7 +16,11 @@ def sync
     end
 
     def deploy_succeeded?
-      @num_endpoints > 0
+      if related_deployment_replicas
```

Comment:
I like the explicitness of `exposes_zero_replica_deployment?` over the current more implicit implementation